src/device/socket/vsock.rs

```diff
@@ -411,11 +411,16 @@ impl<H: Hal, T: Transport> VirtIOSocket<H, T> {
     }
 
     fn send_packet_to_tx_queue(&mut self, header: &VirtioVsockHdr, buffer: &[u8]) -> Result {
-        let _len = self.tx.add_notify_wait_pop(
-            &[header.as_bytes(), buffer],
-            &mut [],
-            &mut self.transport,
-        )?;
+        let _len = if buffer.is_empty() {
```

Comment:
Why do we need `let _len =` instead of using `if ... else` directly?

Reply:
It doesn't do anything, it just makes it explicit that we are ignoring the returned length. We had this before and it seemed worthwhile to keep.